### DISCUSSION TITLE
Write variant columns using SnowflakeIOManager

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#17504] [Q&A]
Hi, we want to write some variant columns using the SnowflakeIOManager as we are loading in data containing JSON. It seems that objects get written as varchar by default. Is there a way to override that?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1698654146256939?thread_ts=1698654146.256939&cid=C01U954MEER).*

**Comment by yuhan** [comment]
Dug in the code implementation [here](https://github.com/dagster-io/dagster/blob/074ae4579f3dde780e78e92414eceb44e58862e6/python_modules/libraries/dagster-snowflake-pandas/dagster_snowflake_pandas/snowflake_pandas_type_handler.py#L66-L72), it seems like we currently don't expose a way to overwrite `dtype` param to the `to_sql` method. cc @jamiedemaria to confirm

Meanwhile, in order to get full control over, I'd suggest switching to use SnowflakeResource. Here's an example: https://docs.dagster.io/integrations/snowflake/reference#executing-custom-sql-commands-with-the-snowflake-resource
